### PR TITLE
Sort categories returned by  "getChildren" by position

### PR DESCRIPTION
**Pull request by EliasKotlyar**
Change "getChildren" category method to return categories sorted by position.
Fixes issue https://github.com/OpenMage/magento-lts/issues/350

### CODE REVIEW

File: app/code/core/Mage/Catalog/Model/Resource/Category/Flat.php [hunk 1]
```diff
@@ -1320,6 +1320,8 @@ public function getChildren($category, $recursive = true, $isActive = true)
         if ($isActive) {
             $select->where('is_active = ?', '1');
         }
+        $maintable = $this->getMainStoreTable($category->getStoreId());
```
Comment (LeeSaferite) [inline app/code/core/Mage/Catalog/Model/Resource/Category/Flat.php]: Can you move this further up and use the variable in both locations?

Comment (EliasKotlyar) [reply app/code/core/Mage/Catalog/Model/Resource/Category/Flat.php]: Thanks for your feedback. I have refactored it.

File: app/code/core/Mage/Catalog/Model/Resource/Category.php [hunk 2]
```diff
@@ -728,6 +728,7 @@ public function getChildren($category, $recursive = true)
                 array()
             )
             ->where($checkSql . ' = :scope');
+        $select->order("m.position ASC");
```
Comment (LeeSaferite) [inline app/code/core/Mage/Catalog/Model/Resource/Category.php]: Can we include this as part of the fluent call right after the `where` and drop the `$select`?

File: app/code/core/Mage/Catalog/Model/Resource/Category/Flat.php [hunk 3]
```diff
@@ -1311,15 +1311,17 @@ public function isInRootCategoryList($category)
...
-            ->from($this->getMainStoreTable($category->getStoreId()), 'entity_id')
+            ->from($maintable, 'entity_id')
             ->where('path LIKE ?', "{$category->getPath()}/%");
         if (!$recursive) {
             $select->where('level <= ?', $category->getLevel() + 1);
         }
         if ($isActive) {
             $select->where('is_active = ?', '1');
         }
+        $select->order($maintable.".position ASC");
```
Comment (LeeSaferite) [inline app/code/core/Mage/Catalog/Model/Resource/Category/Flat.php]: Can we include this as part of the fluent call right after the `where` and drop the `$select`? Moving this up above the if statements would be needed.

File: app/code/core/Mage/Catalog/Model/Resource/Category.php (hunk 2 above)
Comment (EliasKotlyar) [reply app/code/core/Mage/Catalog/Model/Resource/Category.php]: Thanks for your support. I have refactored it.

File: app/code/core/Mage/Catalog/Model/Resource/Category/Flat.php (hunk 3 above)
Comment (EliasKotlyar) [reply app/code/core/Mage/Catalog/Model/Resource/Category/Flat.php]: Thanks for your support. I have refactored it.

**Comment by LeeSaferite** [comment]
This still needs a functional review from one of the other members as I do not currently have a good setup to test things (in the middle of moving).

**Comment by sreichel** [comment]
LGTM.

Question. How about adding a `$sortByPositionFlag` like in M2s PR? (https://github.com/magento/magento2/pull/11342)

**Comment by sreichel** [comment]
> This still needs a functional review from one of the other members ...

- it's not used in core
- the change works as intended

I agree with this PR with still would suggest to use an optional flag (#429) so we can enure not breaking anything,

**Pull request merged**